File: Java/src/main/java/QuestionBank/Solution301_400/Solution303.java
package QuestionBank.Solution301_400;

/**
 * @author windows
 */
public class Solution303 {
    class NumArray {
        private int[] res;

        public NumArray(int[] nums) {
            int len = nums.length;
            res = new int[len + 1];
            res[0] = 0;
            int sum = 0;
            for (int i = 0; i < len; i++) {
                res[i + 1] = sum + nums[i];
                sum += nums[i];
            }
        }

        public int sumRange(int left, int right) {
            return res[right + 1] - res[left];
        }

    }


    public static void main(String[] args) {
        Solution303 solution303 = new Solution303();
    }
}

File: C++/src/Solution301_400/Solution303.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution303.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

NumArray::NumArray(vector<int> &nums) {
    int n = nums.size();
    sums.resize(n + 1);
    for (int i = 0; i < n; ++i) {
        sums[i + 1] = sums[i] + nums[i];
    }
}

int NumArray::sumRange(int left, int right) {
    return sums[right + 1] - sums[left];
}
